api/proto/teleport/machineid/v1/bot_instance.proto

```diff
@@ -0,0 +1,121 @@
+// Copyright 2024 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.machineid.v1;
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+import "teleport/header/v1/metadata.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/machineid/v1;machineidv1";
+
+// A BotInstance
+message BotInstance {
+  // The kind of resource represented.
+  string kind = 1;
+  // Differentiates variations of the same kind. All resources should
+  // contain one, even if it is never populated.
+  string sub_kind = 2;
+  // The version of the resource being represented.
+  string version = 3;
+  // Common metadata that all resources share.
+  teleport.header.v1.Metadata metadata = 4;
+  // The configured properties of a BotInstance.
+  BotInstanceSpec spec = 5;
+  // Fields that are set by the server as results of operations. These should
+  // not be modified by users.
+  BotInstanceStatus status = 6;
+}
+
+// BotInstanceSpec contains fields
+message BotInstanceSpec {
+  // The name of the bot associated with this instance.
+  string bot_name = 1;
+  // The unique identifier for this instance.
+  string instance_id = 2;
+  // The desired expiration offset for this bot instance. The expiration will be
+  // calculated from the current time at creation or authentication plus this
+  // value. A nil `ttl` will not expire.
+  google.protobuf.Duration ttl = 3;
```

Comment:
I've made some minor changes here, partly as discussed: `bot_name` and (now) `instance_id` are part of the initial spec so metadata can be generated from it, and the desired TTL is included now, so we can calculate a desired expiration, and update it appropriately on renew/rejoin/etc.